pages/purchase.tsx

```diff
@@ -1,53 +1,61 @@
-import { Inter } from 'next/font/google'
-import { Box, Button, Image, Flex } from '@chakra-ui/react'
-import { useConnection } from '@/packages/providers'
-import { useEffect } from 'react'
-import { useRouter } from 'next/router'
+import { Inter } from "next/font/google";
+import { Box, Button, Image, Flex } from "@chakra-ui/react";
+import { useConnection } from "@/packages/providers";
+import { useEffect } from "react";
+import { useRouter } from "next/router";
 
-const inter = Inter({ subsets: ['latin'] })
+const inter = Inter({ subsets: ["latin"] });
 
 export default function Home() {
-  const {address, callcontract} = useConnection()
-  const router = useRouter()
-  
+  const { address, callcontract } = useConnection();
+  const router = useRouter();
+
   useEffect(() => {
     if (!address) {
       router.push("/");
     }
-  }, [address])
+  }, [address, router]);
 
-  const purchase = async () => {    
+  const purchase = async () => {
     await callcontract?.({
       amountMutez: 1,
       contractAddress: process.env.NEXT_PUBLIC_CONTRACT_ADDRESS ?? "",
-      id: 'asd'
-
+      id: "asd",
```

Comment:
Any ideas what id stands for here?

Reply:
No I am unsure of what it stands for too